content/java/concepts/hash-map/hash-map.md

```diff
@@ -0,0 +1,216 @@
+    courseTeacher.put("Mathematics", "Jeanette");
+    courseTeacher.put("Physics", "Lily");
+
+    System.out.println(courseTeacher.get("Physics"););
+    // Output: Lily
+
+    System.out.println(courseTeacher.get("History"););
+    // Output: Ben
+  }
+}
+```
+
+## Adding Items
+
+In Java, an item can be added to a `HashMap` using the `put()` method.
+
+The `put()` method takes in 2 attributes. The first attribute is the key and the second attribute is the corresponding value.
+
+```java
+import java.util.HashMap; // import the HashMap class
+
+public class Main {
+  public static void main(String[] args) {
+
+    HashMap<String, String> courseTeacher = new HashMap<String, String>();
+
+    // Add keys and values (CourseNames, Teacher)
+    courseTeacher.put("History", "Ben");
+    courseTeacher.put("Mathematics", "Jeanette");
+    courseTeacher.put("Physics", "Lily");
+
+    System.out.println(courseTeacher); 
+    //Output: {Mathematics=Jeanette, History=Ben, Physics=Lily}
+
+
+  }
+}
+```
+
+## Accessing an Item
+
+In Java, an item can be accessed within a `HashMap` using the `get()` method.
+
+The `get()` method takes in 1 attribute which is the key and returns the corresponding value.
+
+```java
+courseTeacher.get("Physics");
+//Returns "Lily" which is the value for the key "Physics"
+```
+
+## Removing Items
+
+In Java, an item can be removed from a `HashMap` using the `remove()` method.
+
+The `remove()` method takes in 1 attribute which is the key and the method deletes the corresponding key-value pair from the `HashMap`.
+
+```java
+courseTeacher.remove("Physics");
+```
+
+```java
+import java.util.HashMap; // import the HashMap class
+
+public class Main {
+  public static void main(String[] args) {
+
+    HashMap<String, String> courseTeacher = new HashMap<String, String>();
+
+    // put(CourseName, Teacher)
+    courseTeacher.put("History", "Ben");
+    courseTeacher.put("Mathematics", "Jeanette");
+    courseTeacher.put("Physics", "Lily");
+    
+    courseTeacher.remove("Physics");
+    System.out.println(courseTeacher); 
+
+  }
+}
+
+//Output: {Mathematics=Jeanette, History=Ben}
+```
+
+The `clear` method can be used to delete all the items from the `HashMap`.
+
+```java
+courseTeacher.clear();
+```
+
+```java
+import java.util.HashMap; // import the HashMap class
+
+public class Main {
+  public static void main(String[] args) {
+
+    HashMap<String, String> courseTeacher = new HashMap<String, String>();
+
+    // put(CourseName, Teacher)
+    courseTeacher.put("History", "Ben");
+    courseTeacher.put("Mathematics", "Jeanette");
+    courseTeacher.put("Physics", "Lily");
+    
+    courseTeacher.clear();
+    System.out.println(courseTeacher); 
+
+  }
+}
+
+//Output: {}
+```
+
+## Traversing through a HashMap
+
+A `HashMap` can easily be traversed with the for-each loop.
+
+The `keySet()` method can be used to obtain only the keys while the `values()` method can be used to obtain only values.
+
+```java
```

Comment:
```suggestion
```codebyte/java
```

Reply:
I'm thinking this could be a codebyte! If so, we don't need the commented output info.